### PR TITLE
Add 'Guides' section

### PR DESCRIPTION
**Pull request by clstokes**
This makes some minor changes to https://github.com/hashicorp/vault/pull/2217 to be a bit more consistent with HashiCorp's other tools. 

Thanks @rfay for the original PR!

### CODE REVIEW

**Comment by clstokes** [comment]
@jefferai and/or @bensojona look good?

File: website/source/docs/guides/generate-root.html.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+layout: "docs"
+page_title: "Generate Root"
```
Comment (vishalnayak) [inline website/source/docs/guides/generate-root.html.md]: Can we name this as `Generate Root Tokens`?

**Comment by rfay** [comment]
Great stuff - there's room for lots of "guides" in this manual, so this gives us a place. I have a couple more queued up in my notes.

File: website/source/docs/guides/generate-root.html.md (hunk 1 above)
Comment (clstokes) [inline website/source/docs/guides/generate-root.html.md]: Good idea. Done!

**Comment by jefferai** [comment]
@clstokes is there a plan to flesh out the generate-root guide? Right now it doesn't seem worthy of a "guide" designation. If this was just intended as a placeholder to get the Guides section inserted, maybe we can remove that and @rfay can add some stuff in?

**Comment by clstokes** [comment]
@jefferai my intent was that the section is a placeholder for future guides with Randy's being the first. 

I found the generate-root guide helpful as it lays out the order of operations that isn't immediately clear from the `-help` documentation. I personally struggled with this a few weeks ago.

@jefferai what would you like to see in more detail?

**Comment by jefferai** [comment]
@clstokes I think it was just a sort of logical disconnect of me thinking that "Guide" means "something long". Not really an issue!

File: website/source/layouts/docs.erb [hunk 2]
```diff
@@ -104,6 +104,15 @@
 					<a href="/docs/http/index.html">API & Libraries</a>
 				</li>
 
+				<li<%= sidebar_current("docs-guides") %>>
+					<a href="/docs/guides/index.html">Guides</a>
+					<ul class="nav">
+						<li<%= sidebar_current("docs-guides-generate-root") %>>
+							<a href="/docs/guides/generate-root.html">Generate Root</a>
```
Comment (jefferai) [inline website/source/layouts/docs.erb]: I think in the sidebar it'd be better to say "Root Token Generation" here. "Generate Root" is the Vault command but for a newbie it's already slightly inside-baseball. I'll change that and merge it, but if you have strong feelings here, feel free to let me know.

**Comment by jefferai** [comment]
Thanks @rfay and @clstokes !

**Pull request merged**